### PR TITLE
22/dark mode

### LINKED ISSUE #22: Enhancement: finish dark mode theme
### Describe the Current Behavior/Feature:

This issue will need nav bar merged in first to update supporting code. 

With the nav bar is a dark mode toggle. Right now there is basically enough of a dark mode theme to test that it was set up properly. Needs a full dark mode theme using [MUI dark mode colors] (https://mui.com/material-ui/customization/dark-mode/).  

### Proposed Implementation:

All of the magic goes in theme.js

### PR DESCRIPTION
**Pull request by JaeGif**
## This PR:

Resolves #22

Made some blobs to contrast low-contrast items. Open to suggestions on design of the blobs/placement etc. Otherwise this should cover the full dark-mode.

### CODE REVIEW

File: src/Layout.jsx [hunk 1]
```diff
@@ -24,5 +26,8 @@ const Layout = () => {
     </>
   );
 };
-
+Layout.propTypes = {
```
Comment (joncheng-dev) [inline src/Layout.jsx:29]: propTypes potentially deprecated in future; 
https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-proptypes-and-defaultprops

File: src/components/home/ProjectsBrief.jsx [hunk 2]
```diff
@@ -8,74 +8,105 @@ import CardContent from '@mui/material/CardContent';
...
+    p: { xs: 1, sm: 3 },
+    background: `${theme.palette.primary.cardFill}`,
+    borderRadius: '30px',
+    height: { xs: 'auto', sm: '100%' },
+    border:
+      theme.palette.mode === 'dark' ? `1px ${theme.palette.primary.boxOutline} solid` : 'none',
+    minHeight: { sm: '500px' },
+    backgroundImage:
+      theme.palette.mode === 'dark'
+        ? 'url(/assets/partnerLogos/backgroundBlobs/mobileBlob.webp)'
```
Comment (kimmykokonut) [inline src/components/home/ProjectsBrief.jsx:32]: The dark mode green is not the 'St John's Green' that matches the buttons. Desirable to be #56b6a1

File: src/components/global/Navbar.jsx [hunk 3]
```diff
@@ -156,7 +162,11 @@ function NavBar() {
...
         <Typography component="h1" variant="h4" display={{ xs: 'none', sm: 'block' }} pl={'10px'}>
-          <Link to="/" style={navTextStyle} aria-label="CODE PDX">
+          <Link
+            to="/"
+            style={{ ...navTextStyle, color: theme.palette.primary.contrastText }}
+            aria-label="CODE PDX"
+          >
             CODE PDX
           </Link>
         </Typography>
```
Comment (kimmykokonut) [inline src/components/global/Navbar.jsx:172]: The 'CODE PDX' (and other nav elements in header) are now black on light mode but should remain white as in current website. (But maybe there is a case for accessibility? If so, then maybe the rose logo is also black for consistency?)

**Review by Jared-Krajewski — CHANGES_REQUESTED** [review]
Looks great, few small things. 

1. this blob doesn't fully encapsulate the rose logo.
Screenshot 2024-05-29 at 9 55 17 PM

2. All blobs are slightly off in color (I'm hopeful you have the original files and can just update the fill color, again thanks for making these) should be #56B6A1, same color code mentioned in other comments. 

3. In regards to the nav bar text being black, I am ok with this staying black for now since white failed contrast standards. I can do another PR to update the rose logo.

File: src/components/global/Navbar.jsx (hunk 3 above)
Comment (Jared-Krajewski) [reply src/components/global/Navbar.jsx:172]: Ok with this change as white failed contrast standards so this is better for now.

**Comment by JaeGif** [comment]
I do have the original file and will do some tweaking. The color code is actually correct in photoshop so it seems to be some artifact involved with getting a webp version. Will look into solutions tonight.

**Comment by JaeGif** [comment]
Changed blob colors (still slightly off from the footer due to the gradient) but matching the buttons colors.

Made the blob encapsulating the rose bigger to fit the whole rose.

**Review by Jared-Krajewski — APPROVED** [review]
awesome work, thanks for doing all this! I can re do the footer in another PR to fix that.

**Pull request merged**